tests/test_encoding.py

```diff
@@ -1,80 +1,103 @@
+        assert self.kind(codec.decode(1)) == 'color_accepting'
+        assert self.kind(codec.decode(3)) == 'color_accepting'
+        assert self.kind(codec.decode(4)) == 'color_node'
+        assert self.kind(codec.decode(33)) == 'color_node'
+        assert self.kind(codec.decode(34)) == 'parent_relation'
 
-    colors = range(codec.n_colors)
-    nodes = range(codec.n_nodes)
-    tokens = range(codec.n_tokens)
+        colors = range(codec.n_colors)
+        nodes = range(codec.n_nodes)
+        tokens = range(codec.n_tokens)
 
-    # Test bijection on color accepting vars.
-    lits = set()
-    for c in colors:
-        lit = codec.color_accepting(c)
-        lits.add(lit)
-        var = codec.decode(lit)
-        assert kind(var) == 'color_accepting'
-        assert var.color == c
-    assert len(lits) == 3
+        # Test bijection on color accepting vars.
+        lits = set()
+        for c in colors:
+            lit = codec.color_accepting(c)
+            lits.add(lit)
+            var = codec.decode(lit)
+            assert self.kind(var) == 'color_accepting'
+            assert var.color == c
+        assert len(lits) == 3
 
-    # Test bijection on color nodes.
-    lits = set()
-    for n, c in product(nodes, colors):
-        lit = codec.color_node(n, c)
-        lits.add(lit)
-        var = codec.decode(lit)
-        assert kind(var) == 'color_node'
-        assert var.color == c
-        assert var.node == n
-    assert len(lits) == 3 * 10  # Check bijection.
+        # Test bijection on color nodes.
+        lits = set()
+        for n, c in product(nodes, colors):
+            lit = codec.color_node(n, c)
+            lits.add(lit)
+            var = codec.decode(lit)
+            assert self.kind(var) == 'color_node'
+            assert var.color == c
+            assert var.node == n
+        assert len(lits) == 3 * 10  # Check bijection.
 
-    # Test bijection on parent relation vars.
-    lits = set()
-    for t, c1, c2 in product(tokens, colors, colors):
-        lit = codec.parent_relation(t, c1, c2)
-        lits.add(lit)
-        var = codec.decode(lit)
-        assert kind(var) == 'parent_relation'
-        assert var.parent_color == c1
-        assert var.node_color == c2
-        assert var.token == t
+        # Test bijection on parent relation vars.
+        lits = set()
+        for t, c1, c2 in product(tokens, colors, colors):
+            lit = codec.parent_relation(t, c1, c2)
+            lits.add(lit)
+            var = codec.decode(lit)
+            assert self.kind(var) == 'parent_relation'
+            assert var.parent_color == c1
+            assert var.node_color == c2
+            assert var.token == t
 
-    assert len(lits) == 9 * 4  # Check bijection.
-    assert max(lits) == 3 + 3 * 10 + 9 * 4
+        assert len(lits) == 9 * 4  # Check bijection.
+        assert max(lits) == 3 + 3 * 10 + 9 * 4
 
+    def test_symm_break(self):
```

Comment:
Also, can we also test if symmetry break is correct by checking that there is at most 1 model.

We could do this a number of ways, but my proposal would be to take the codec and solve for the model as in the identify function.

Then assert that that particular model is not allowed by adding a clause with the negated literals.

`diff_model = [-lit for lit in model]`

Adding that clause to the SAT solver should make the instance unsat.